core/sdk/src/api/firebolt/fb_discovery.rs

```diff
@@ -61,10 +63,19 @@ impl LaunchRequest {
 #[derive(Debug, Serialize, Deserialize, Clone, Default)]
 #[serde(rename_all = "camelCase")]
 pub struct EntitlementData {
+    #[serde(deserialize_with = "valid_string_deserializer")]
     pub entitlement_id: String,
```

Comment:
Is empty string not a valid input value for `entitlement_id` ?  Content Access is mainly a pass through feature. 
What if one of the distributors would like  to collect empty string as `entitlement_id` for some specific reason.

Reply:
Am not sure too feels odd to not have an id given that start_time and end_time are optional too.

Reply:
will revert this change